### PR TITLE
Set SO_LINGER in integration tests.

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

We now have sufficiently many integration tests that the number of TCP
sockets in TIME_WAIT begins to cause resource pressure on the system.
This can lead to the integration tests failing due to being unable to
assign a socket.

We can force this to not happen by disabling linger. As we're running on
a lossless fabric (localhost), this is entirely safe.

Modifications:

- Disable TIME_WAIT for integration tests.

Result:

Integration tests will pass.

### CODE REVIEW

**Comment by Lukasa** [comment]
@weissi please re-review in the context of the new allocation counter changes.

File: docker/docker-compose.1604.53.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ services:
       - MAX_ALLOCS_ALLOWED_udp_1000_reqs_1_conn=16250
       - MAX_ALLOCS_ALLOWED_udp_1_reqs_1000_conn=226050
       - MAX_ALLOCS_ALLOWED_1000_tcpbootstraps=4100
-      - MAX_ALLOCS_ALLOWED_1000_tcpconnections=208100
+      - MAX_ALLOCS_ALLOWED_1000_tcpconnections=223100
```
Comment (weissi) [inline docker/docker-compose.1604.53.yaml:39]: that's interesting. Why is this +15 allocs and the test case that nominally does _more (`1_reqs_1000_conn`) only gains + 1? Any ideas @PeterAdams-A ?

Comment (weissi) [reply docker/docker-compose.1604.53.yaml:39]: ping @Lukasa / @PeterAdams-A

**Comment by PeterAdams-A** [comment]
@swift-nio-bot test this please

**Comment by PeterAdams-A** [comment]
@swift-nio-bot please test this

**Comment by PeterAdams-A** [comment]
@swift-nio-bot test this please

File: docker/docker-compose.1604.53.yaml (hunk 1 above)
Comment (PeterAdams-A) [reply docker/docker-compose.1604.53.yaml]: It's now +2 vs +1.

File: docker/docker-compose.1604.52.yaml [hunk 2]
```diff
@@ -37,7 +37,7 @@ services:
       - MAX_ALLOCS_ALLOWED_udp_1000_reqs_1_conn=16250
       - MAX_ALLOCS_ALLOWED_udp_1_reqs_1000_conn=226050
       - MAX_ALLOCS_ALLOWED_1000_tcpbootstraps=4100
-      - MAX_ALLOCS_ALLOWED_1000_tcpconnections=184010
+      - MAX_ALLOCS_ALLOWED_1000_tcpconnections=180010
```
Comment (weissi) [inline docker/docker-compose.1604.52.yaml:40]: -4 ? haha

File: IntegrationTests/tests_04_performance/test_01_resources/test_1000_tcpconnections.swift [hunk 3]
```diff
@@ -29,6 +29,19 @@ fileprivate final class ReceiveAndCloseHandler: ChannelInboundHandler {
     }
 }
 
+fileprivate final class LingerSettingHandler: ChannelInboundHandler {
+    typealias InboundIn = ByteBuffer
+    typealias OutboundOut = ByteBuffer
+    
+    public func handlerAdded(context: ChannelHandlerContext) {
+        _ = (context.channel as? SocketOptionProvider)?.setSoLinger(linger(l_onoff: 1, l_linger: 0))
```
Comment (weissi) [inline IntegrationTests/tests_04_performance/test_01_resources/test_1000_tcpconnections.swift:37]: shouldn't we blow up in a `.whenFailure { ... }` here?

Comment (Lukasa) [reply IntegrationTests/tests_04_performance/test_01_resources/test_1000_tcpconnections.swift:37]: Can't hurt. @PeterAdams-A as you've adopted this branch, want to keep going?

File: docker/docker-compose.1604.52.yaml (hunk 2 above)
Comment (PeterAdams-A) [reply docker/docker-compose.1604.52.yaml:40]: Where do you see -4?

File: IntegrationTests/tests_04_performance/test_01_resources/test_1000_tcpconnections.swift (hunk 3 above)
Comment (PeterAdams-A) [reply IntegrationTests/tests_04_performance/test_01_resources/test_1000_tcpconnections.swift]: Done this.

**Pull request merged**